lib/common/xml.c

```diff
@@ -849,7 +850,16 @@ string2xml(const char *input)
     xmlCtxtResetLastError(ctxt);
     xmlSetGenericErrorFunc(ctxt, pcmk__log_xmllib_err);
     output = xmlCtxtReadDoc(ctxt, (pcmkXmlStr) input, NULL, NULL,
-                            PCMK__XML_PARSE_OPTS);
+                            PCMK__XML_PARSE_OPTS_WITHOUT_RECOVER);
+
+    if (output == NULL) {
+        output = xmlCtxtReadDoc(ctxt, (pcmkXmlStr) input, NULL, NULL,
+                                PCMK__XML_PARSE_OPTS_WITH_RECOVER);
+        if (output) {
+            crm_warn("Successfully recovered from XML errors. Note: a future release will treat this as a fatal failure.");
```

Comment:
You can use string literal concatenation to keep these under 80 characters, for example

    crm_warn("Successfully recovered from XML errors "
             "(note: a future release will treat this as a fatal failure)");